controller.go

```diff
@@ -837,10 +837,24 @@ addToStore:
 	if err = c.updateToStore(network); err != nil {
 		return nil, err
 	}
+	defer func() {
+		if err != nil {
+			if e := c.deleteFromStore(network); e != nil {
+				logrus.Warnf("could not rollback from store, network %v on failure (%v): %v", network, err, e)
+			}
+		}
+	}()
+
 	if network.configOnly {
 		return network, nil
 	}
 
+	if len(network.loadBalancerIP) != 0 {
```

Comment:
@pradipd apologies on commenting on this after my LGTM. But I noticed that this block should be moved to the end of the function. Otherwise, it can potentially cause a Gossip timing issue since `createLoadBalancerSandbox` creates endpoints.. But the `joinCluster` is done after the endpoint is created. `joinCluster` is the function that creates the required networkDB states for a network and it is better if the gossip network states are created prior to pushing endpoint events into it.